system/jlib/jthread.cpp

```diff
@@ -1017,19 +1019,25 @@ class CThreadPool: public CThreadPoolBase, implements IThreadPool, public CInter
     PooledThreadHandle _start(void *param,const char *name, bool noBlock, unsigned timeout=0)
     {
         CCycleTimer startTimer;
-        bool waited = false;
-        bool timedout = false;
+        bool waited{false};
+        bool timedout{false};
+        bool slotConsumed{true};
         if (defaultmax)
         {
             waited = !availsem.wait(0);
             if (noBlock)
                 timedout = waited;
             else if (waited)
+            {
                 timedout = !availsem.wait(timeout>0?timeout:delay);
+                if (timedout)
+                    slotConsumed = false;
```

Comment:
this would be incorrect, if the line inside the if (timedout) in the CS block below consumed the signal.
i.e. it timedout, it will fall into that block, and if !availsem.wait(0) , the slotConsumed = false there will set and shouldn't set here.